tools/check-branch

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+set -e
+
+if ! git diff-index --quiet HEAD; then
+    set +x
+    echo "There are uncommitted changes:"
+    git status --short
+    echo "Doing nothing to avoid losing your work."
+    exit 1
+fi
+
+main_branch=${1:-"upstream/main"}
+
+if [ ! "$1" ]; then
+    echo "Using default main branch '$main_branch'"
+    if ! git diff --quiet "$main_branch" main; then
+        set +x
+        echo "Branch '$main_branch' appears to differ from 'main'; please clarify!"
+        exit 1
+    fi
+else
+    echo "Specified main branch as '$main_branch'"
+fi
+
+if git diff --quiet "$main_branch" HEAD; then
+    set +x
+    echo "No changes! Nothing to do!"
+    exit 1
+fi
+
+echo "Checking commits..."
+git rebase $(git merge-base "$main_branch" @) --exec 'git log -1 && make check'
```

Comment:
`git log -1` make the process stop at each commit until I press `q`. Was that intended? What about using `cat <(git log -1)` instead?

Reply:
I tried using `--no-pager` instead; does that resolve this issue for you?
(this avoids extra commands/shells, though I agree various options are possible)